### ISSUE TITLE
Analysis server fails to notice that a file re-exporting `main` is runnable

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#45192]
Steps to reproduce:
- Check out Dart SDK as of 8be535350debe305e9ccab60399db3c02976f30c
- In IntelliJ, open the file `pkg/front_end/test/unit_test_suites.dart`
- Right click within the file view

Expected result: the menu item `Run 'unit_test_suites.dart'` should be shown

Observed result: no `Run` menu item is shown

Note that all this file does is export `unit_test_suites_impl.dart`, which declares a `main` function.  Therefore it should have a `main` function in its export namespace, so it should be runnable.  Prior to null safety, this would have been a really obscure use case.  However, with null safety, there's actually a really good reason to do this: it allows the entry point to be marked with `// @dart = 2.9` (so that the program will be run in weak null safety mode), while leaving `main` opted in (so that it gets the static compilation benefits of null safety mode).  This is necessary because due to out-of-order migration, some of the code in the program's transitive import graph has not yet been migrated to null safety.  Now that null safety is out, I suspect this is going to be a much more common use case.

**Comment by bwilkerson** [comment]
@alexander-doroshko I don't remember having a server protocol to tell clients whether a file is executable, so I suspect that this is being done in the IntelliJ plugin. Can you confirm?

**Comment by alexander-doroshko** [comment]
True, run configurations are handled by the Dart plugin. We don't handle such case yet. Request is welcome: https://youtrack.jetbrains.com/issues/WEB (Project: Web, Subsystem: Dart).

For now, you'll need to configure a run configuration manually: main menu -> Run -> Edit Configurations -> [+] -> Dart Test
